app/routes/upload.step1.tsx

```diff
@@ -0,0 +1,95 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { Form, redirect, useNavigation } from '@remix-run/react'
+import { useState } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import { Button } from '~/components/ui/button'
+import { supabase } from '~/supabaseClient'
+
+// Action to handle form submission
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const body = await request.formData()
+
+  const file = body.get('cv-file') as File | null
+
+  if (file) {
+    const uniqueFileName = `${uuidv4()}-${file.name}`
+    const { data, error } = await supabase.storage
+      .from('resumes')
+      .upload(`public/${uniqueFileName}`, file) // this is just for testing now. will be secured later
+
+    if (error) {
+      console.error(error)
+    } else {
```

Comment:
**issue**: Error handling is currently limited to console logging. Consider providing user feedback for upload errors.